domain/src/main/java/com/example/domain/repository/CharacterRepository.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.domain.repository
+
+import com.example.domain.models.CharacterSchema
+import com.example.domain.utils.Results
+import kotlinx.coroutines.flow.Flow
+
+interface CharacterRepository {
+    suspend fun getCharacters(): Flow<Results<List<CharacterSchema?>>>
```

Comment:
The `domain` layer is supposed to be the blueprint of the whole system. And a blueprint should be as simple as possible. Since the function `getCharacters` is not supposed to `observe` the characters and only fetch them once, make sure to give this function a very simple signature, like:

```kotlin
suspend fun getCharacters(): List<Character>
```

> [!NOTE]
Notice how I'm simplifying the name of the model class as well (from `CharacterSchema` to `Character`) and how easy it has now become to understand the functionality of this method without actually having to dive into it.